src/useLoadBans.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect, type Dispatch } from "react";
+import { type Action } from "./useAppState";
```

Comment:
when all the imports are type imports we can also do this

```suggestion
import type { Action } from "./useAppState";
```